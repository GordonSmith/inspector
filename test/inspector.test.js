/* eslint-env jest */
var Inspector = require("../dist/notebook-inspector.umd.js").Inspector;

describe("basic inspector", () => {
  let inspector;
  beforeEach(() => {
    const elem = document.createElement("div");
    inspector = new Inspector(elem);
  });

  test("initial state", () => {
    expect(inspector).toMatchSnapshot();
  });

  test("pending state", () => {
    inspector.pending();
    expect(inspector).toMatchSnapshot();
  });

  test("fulfilled with an element", () => {
    const span = document.createElement("span");
    span.textContent = "Surprise!";
    inspector.fulfilled(span);
    expect(inspector).toMatchSnapshot();
  });

  test("fulfilled with a complex vlaue", () => {
    inspector.fulfilled([1, 2, 3]);
    expect(inspector).toMatchSnapshot();
  });

  test("rejected state", () => {
    inspector.rejected(new Error("Danger!"));
    expect(inspector).toMatchSnapshot();
  });
});
