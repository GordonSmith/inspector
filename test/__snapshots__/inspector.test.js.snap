// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inspector .fulfilled(element) 1`] = `
R {
  "_node": <div
    class="observablehq"
  >
    <span>
      Surprise!
    </span>
  </div>,
}
`;

exports[`Inspector .fulfilled(value) 1`] = `
R {
  "_node": <div
    class="observablehq"
  >
    <span
      class="observablehq--collapsed observablehq--inspect"
    >
      <a>
        â–¸Array(3) [
      </a>
      <span
        class="observablehq--number"
      >
        1
      </span>
      , 
      <span
        class="observablehq--number"
      >
        2
      </span>
      , 
      <span
        class="observablehq--number"
      >
        3
      </span>
      ]
    </span>
  </div>,
}
`;

exports[`Inspector .pending() 1`] = `
R {
  "_node": <div
    class="observablehq observablehq--running"
  />,
}
`;

exports[`Inspector .rejected(Error) 1`] = `
R {
  "_node": <div
    class="observablehq observablehq--error"
  >
    <span
      class="observablehq--inspect"
    >
      Error: Danger!
    </span>
  </div>,
}
`;

exports[`Inspector initial state 1`] = `
R {
  "_node": <div
    class="observablehq"
  />,
}
`;

exports[`into into(div) 1`] = `
<div>
  <div
    class="observablehq"
  >
    <span
      class="observablehq--number observablehq--inspect"
    >
      42
    </span>
  </div>
</div>
`;
